From 88dfcc35c499ed66fc98f50eaecf7291433e27d2 Mon Sep 17 00:00:00 2001
From: Murat Sezgin <msezgin@codeaurora.org>
Date: Wed, 2 Mar 2016 11:12:59 -0800
Subject: [PATCH] Pass MAC address to Hy-Fi bridge

- This may be a skb on the forward or reverse direction of the
  connection
- Need to pass the dest address as well as the skb to determine if we
  need to calculate the forward or reverse hash

Change-Id: I245855910ac91f73a9e9e6bd3403f01bc662a93a
Signed-off-by: Murat Sezgin <msezgin@codeaurora.org>
---
 include/linux/if_bridge.h | 3 ++-
 net/bridge/br_if.c        | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

--- a/include/linux/if_bridge.h
+++ b/include/linux/if_bridge.h
@@ -205,7 +205,8 @@ static inline clock_t br_get_ageing_time
 #endif
 
 typedef struct net_bridge_port *br_port_dev_get_hook_t(struct net_device *dev,
-		struct sk_buff *skb);
+						       struct sk_buff *skb,
+						       unsigned char *addr);
 extern br_port_dev_get_hook_t __rcu *br_port_dev_get_hook;
 
 typedef void (br_notify_hook_t)(int group, int event, const void *ptr);
--- a/net/bridge/br_if.c
+++ b/net/bridge/br_if.c
@@ -829,7 +829,8 @@ struct net_device *br_port_dev_get(struc
 		port_dev_get_hook = rcu_dereference(br_port_dev_get_hook);
 		if (port_dev_get_hook) {
 			struct net_bridge_port *pdst =
-				__br_get(port_dev_get_hook, NULL, dev, skb);
+				__br_get(port_dev_get_hook, NULL, dev,
+					skb, addr);
 			if (pdst) {
 				dev_hold(pdst->dev);
 				netdev = pdst->dev;
